java/src/jmri/implementation/AbstractSignalHead.java

```diff
@@ -201,8 +201,6 @@ public void vetoableChange(java.beans.PropertyChangeEvent evt) throws java.beans
                 java.beans.PropertyChangeEvent e = new java.beans.PropertyChangeEvent(this, "DoNotDelete", null, null);
                 throw new java.beans.PropertyVetoException(Bundle.getMessage("InUseTurnoutSignalHeadVeto", getDisplayName()), e); //IN18N
             }
-        } else if ("DoDelete".equals(evt.getPropertyName())) {
-            log.warn("not clear DoDelete operated? {}", getSystemName()); //NOI18N
```

Comment:
Doesn’t this remove the warning for any turnout, not just unrelated ones?

I may have missed it, but I don’t see anything in the event handler that actually checks to see if the turnout is related or not.

Reply:
Deletion of related turnouts is vetoed in the CanDelete event.  This means any turnout identified in the DoDelete event must be unrelated.  Or there is a bug where DoDelete is called without a CanDelete or the veto was ignored.